platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/DataFileCompactor.java

```diff
@@ -0,0 +1,476 @@
+                // It will be removed once the solution comes with #7501 is implemented. The plan is to have a
+                // compaction list.
+                // That is, before we run a compaction we create a list of files from (possibly) all levels to be
+                // compacted.
+                // See https://www.notion.so/swirldslabs/Compaction-Improvements-247726614d924fbaa34aa82a157a2f20 for
+                // details
+
+                readersOfLevel(1)
+                                .apply((List<DataFileReader<?>>) allCompactableFiles)
+                                .size()
+                        > MAX_FIRST_LEVEL_FILES_ALLOWED) {
+            lastMediumMerge = timestamp;
+            filesToCompactFilter = readersOfLevel(1);
+            compactionLevel = CompactionType.MEDIUM;
+            logger.info(MERKLE_DB.getMarker(), "[{}] Starting Medium Merge", storeName);
+        } else {
+            filesToCompactFilter = readersOfLevel(0);
+            compactionLevel = CompactionType.SMALL;
+            logger.info(MERKLE_DB.getMarker(), "[{}] Starting Small Merge", storeName);
+        }
+
+        final List<DataFileReader<?>> filesToCompact =
+                filesToCompactFilter.apply((List<DataFileReader<?>>) allCompactableFiles);
+        if (filesToCompact == null) {
+            // nothing to do
+            return;
+        }
+        final int filesCount = filesToCompact.size();
+        if (filesCount < getMinNumberOfFilesToMerge()) {
+            logger.debug(
+                    MERKLE_DB.getMarker(),
+                    "[{}] No need to merge as {} is less than the minimum {} files to merge.",
+                    storeName,
+                    filesCount,
+                    getMinNumberOfFilesToMerge());
+            return;
+        }
+
+        final long start = System.currentTimeMillis();
+
+        final long filesToMergeSize = getSizeOfFiles(filesToCompact);
+        logger.debug(
+                MERKLE_DB.getMarker(),
+                "[{}] Starting merging {} files / {}",
+                storeName,
+                filesCount,
+                formatSizeBytes(filesToMergeSize));
+
+        final List<Path> newFilesCreated = compactFiles(index, filesToCompact);
+
+        final long end = System.currentTimeMillis();
+        final long tookMillis = end - start;
+        if (reportDurationMetricFunction != null) {
+            reportDurationMetricFunction.accept(compactionLevel, tookMillis);
+        }
+
+        final long mergedFilesSize = getSizeOfFilesByPath(newFilesCreated);
+        if (reportSavedSpaceMetricFunction != null) {
+            reportSavedSpaceMetricFunction.accept(
+                    compactionLevel, (filesToMergeSize - mergedFilesSize) * UnitConstants.BYTES_TO_MEBIBYTES);
+        }
+
+        logMergeStats(storeName, tookMillis, filesToCompact, filesToMergeSize, newFilesCreated, dataFileCollection);
```

Comment:
`logMergeStats`' second parameter is `tookSeconds`. Passing a value in milliseconds distorts all metrics.

Reply:
I updated `logMergeStats` to convert milliseconds to seconds.